compositor_pipeline/src/pipeline.rs

```diff
@@ -26,11 +27,17 @@ use crate::error::{
 };
 use crate::queue::Queue;
 
-pub trait PipelineOutput: Send + Sync + 'static {
-    type Opts;
+use self::encoder::{Encoder, EncoderSettings};
+
+pub mod encoder;
+
+pub trait PipelineOutputReceiver: 'static {
```

Comment:
I find this name slightly confusing. It's a `Receiver` but it has a `send` method and no `receive` method